app/services/update_wiki_namespace_stats.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class UpdateWikiNamespaceStats
+  def initialize(course, wiki, namespace)
+    @course = course
+    @wiki = wiki
+    @namespace = namespace
+    update_stats
+  end
+
+  def update_stats
+    stat_key = "#{@wiki.domain}-namespace-#{@namespace}"
+    stats = {
+      'edited_count': edited_articles_count,
+      'new_count': new_articles_count,
+      'revision_count': revision_count,
+      'user_count': user_count,
+      'word_count': word_count,
+      'reference_count': reference_count,
+      'view_count': view_count
+    }
+    course_stats = CourseStat.find_or_create_by(course_id: @course.id)
+    course_stats.stats_hash[stat_key] = stats
+    course_stats.save
+  end
+
+  # live, tracked course revisions filtered by wiki and namespace
+  def revisions_filtered_by_wiki_namespace
+    @course.revisions.where.not(article_id: @course.articles_courses.not_tracked.pluck(:article_id))
+                     .where(wiki_id: @wiki.id)
+                     .namespace(@namespace).live
+  end
+
+  # live, tracked articles filtered by wiki and namespace
+  def articles_filtered_by_wiki_namespace
+    @course.articles_courses.joins(:article).where(articles: { wiki: @wiki, namespace: @namespace })
```

Comment:
This doesn't include generating ArticlesCourses for non-mainspace articles, correct? So this will always be zero for this PR?

Reply:
Oh right. I missed it. Will correct it.

Reply:
@ragesoss, right now tracked_namespaces are not known, so should I enable article_courses for all namespaces?

Reply:
no, i don't think so. I think it would be better to include a minimal version of tracked_namespaces in this PR such that it could be deployed without changing current behavior, but would still allow for testing.

One approach would be to define `tracked_namespaces` for `course.rb` but just have it return `[0]` (ie, hard-code the current behavior of mainspace being the only tracked namespace for any course), and then in the test you can stub that method to override it and add cookbook. This would be the simplest way i could think of for this PR, although it would mean a small amount of rewriting for the next stage.

Another approach would be include the new database table in this PR, so that tracked_namespaces could be determined from the database (which will be necessary for the later PRs anyway).

Reply:
But, are you going to deploy it right away? I mean before we have `tracked_namespaces` from `CourseWikiNamespaces` ?

Reply:
Maybe! Part of the point of the separate PRs is to make it possible to deploy before everything else is ready.